typescript/sdk/src/providers/transactions/transformer/ethersV5/EV5InterchainAccountTxTransformer.ts

```diff
@@ -27,31 +21,30 @@ export class EV5InterchainAccountTxTransformer
 
   constructor(
     public readonly multiProvider: MultiProvider,
-    public readonly chain: ChainName,
     public readonly props: EV5InterchainAccountTxTransformerProps,
   ) {}
 
   public async transform(
     ...txs: PopulatedTransaction[]
   ): Promise<PopulatedTransaction[]> {
-    const destinationChainId = txs[0].chainId;
-    assert(
-      destinationChainId,
-      'Missing destination chainId in PopulatedTransaction.',
-    );
-
     const innerCalls: CallData[] = txs.map(
-      ({ to, data, value }: PopulatedTransaction) => {
+      ({ to, data, value, chainId }: PopulatedTransaction) => {
         assert(to, 'Invalid PopulatedTransaction: Missing to field');
         assert(data, 'Invalid PopulatedTransaction: Missing data field');
+        assert(chainId, 'Invalid PopulatedTransaction: Missing chainId field');
+        const txChain = this.multiProvider.getChainName(chainId);
+        assert(
+          txChain === this.props.chain,
+          `Invalid PopulatedTransaction: Cannot submit ${txChain} tx to ${this.props.chain} submitter.`,
+        );
         return { to, data, value };
       },
     );
 
     return [
       await this.props.interchainAccount.getCallRemote(
-        this.chain,
-        this.multiProvider.getChainName(this.chain), //chainIdToMetadata[destinationChainId].name,
+        this.props.chain,
+        this.multiProvider.getChainName(txs[0].chainId!),
```

Comment:
I suppose we should assert that the chain Id is consistent, or actually produce separate callRemote bundles for different chain Ids